samples/boards/nrf/nrfx/src/main.c

```diff
@@ -41,78 +44,95 @@ void main(void)
 	 */
 	err = nrfx_gpiote_init(0);
 	if (err != NRFX_SUCCESS) {
-		LOG_ERR("nrfx_gpiote_init error: %08x", err);
+		LOG_ERR("nrfx_gpiote_init error: 0x%08X", err);
 		return;
 	}
 
-	nrfx_gpiote_in_config_t const in_config = {
-		.sense = NRF_GPIOTE_POLARITY_HITOLO,
-		.pull = NRF_GPIO_PIN_PULLUP,
-		.is_watcher = false,
-		.hi_accuracy = true,
-		.skip_gpio_setup = false,
-	};
+	err = nrfx_gpiote_channel_alloc(&in_channel);
+	if (err != NRFX_SUCCESS) {
+		LOG_ERR("Failed to allocate in_channel, error: 0x%08X", err);
+		return;
+	}
+
+	err = nrfx_gpiote_channel_alloc(&out_channel);
+	if (err != NRFX_SUCCESS) {
+		LOG_ERR("Failed to allocate out_channel, error: 0x%08X", err);
+		return;
+	}
 
 	/* Initialize input pin to generate event on high to low transition
 	 * (falling edge) and call button_handler()
 	 */
-	err = nrfx_gpiote_in_init(INPUT_PIN, &in_config, button_handler);
+	static const nrfx_gpiote_input_config_t input_config = {
+		.pull = NRF_GPIO_PIN_PULLUP,
+	};
+	const nrfx_gpiote_trigger_config_t trigger_config = {
+		.trigger = NRFX_GPIOTE_TRIGGER_HITOLO,
+		.p_in_channel = &in_channel,
+	};
+	static const nrfx_gpiote_handler_config_t handler_config = {
+		.handler = button_handler,
+	};
+	err = nrfx_gpiote_input_configure(INPUT_PIN,
+					  &input_config,
+					  &trigger_config,
+					  &handler_config);
 	if (err != NRFX_SUCCESS) {
-		LOG_ERR("nrfx_gpiote_in_init error: %08x", err);
+		LOG_ERR("nrfx_gpiote_input_configure error: 0x%08X", err);
 		return;
 	}
 
-	nrfx_gpiote_out_config_t const out_config = {
-		.action = NRF_GPIOTE_POLARITY_TOGGLE,
-		.init_state = 1,
-		.task_pin = true,
-	};
-
 	/* Initialize output pin. SET task will turn the LED on,
 	 * CLR will turn it off and OUT will toggle it.
 	 */
-	err = nrfx_gpiote_out_init(OUTPUT_PIN, &out_config);
+	static const nrfx_gpiote_output_config_t output_config = {
+		.drive = NRF_GPIO_PIN_S0S1,
+		.input_connect = NRF_GPIO_PIN_INPUT_DISCONNECT,
+		.pull = NRF_GPIO_PIN_NOPULL,
+	};
+	const nrfx_gpiote_task_config_t task_config = {
+		.task_ch = out_channel,
+		.polarity = NRF_GPIOTE_POLARITY_TOGGLE,
+		.init_val = 1,
+	};
+	err = nrfx_gpiote_output_configure(OUTPUT_PIN,
+					   &output_config,
+					   &task_config);
 	if (err != NRFX_SUCCESS) {
-		LOG_ERR("nrfx_gpiote_out_init error: %08x", err);
+		LOG_ERR("nrfx_gpiote_output_configure error: 0x%08X", err);
 		return;
 	}
 
-	nrfx_gpiote_in_event_enable(INPUT_PIN, true);
+	nrfx_gpiote_trigger_enable(INPUT_PIN, true);
 	nrfx_gpiote_out_task_enable(OUTPUT_PIN);
 
 	LOG_INF("nrfx_gpiote initialized");
 
 	/* Allocate a (D)PPI channel. */
 #if defined(DPPI_PRESENT)
-	uint8_t channel;
-	err = nrfx_dppi_channel_alloc(&channel);
+	uint8_t ppi_channel;
+
+	err = nrfx_dppi_channel_alloc(&ppi_channel);
```

Comment:
sorry for not spotting this earlier, but we have `nrfx_gppi_channel_alloc` as well

Reply:
Applied.